notify/telegram/telegram.go

```diff
@@ -44,6 +45,19 @@ func New(conf *config.TelegramConfig, t *template.Template, l log.Logger, httpOp
 		return nil, err
 	}
 
+	if conf.APIUrl == nil {
+		apiURL, err := url.Parse("https://api.telegram.org")
+		if err != nil {
+			return nil, err
+		}
+		conf.APIUrl = &config.URL{URL: apiURL}
+	}
+	if conf.ParseMode == "" {
+		// The default Telegram template is in HTML,
+		// so we need to set the parse mode to HTML for Telegram not to return a parse error.
+		conf.ParseMode = "HTML"
```

Comment:
This should be done during unmarhalling of the configuration, not when calling New.

Reply:
Our current default is actually `MarkdownV2` this would effectively be a breaking change and I don't think it was the intent of #2827.

Looking at the current template - I don't see how it is HTML, am I missing something? https://github.com/prometheus/alertmanager/blob/main/template/default.tmpl#L105-L114